src/views/AddAddressLedger.vue

```diff
@@ -0,0 +1,112 @@
+            <IdenticonSelector v-else-if="state === 'identicon-selection'" :accounts="addressesToSelectFrom"
+                               :confirmAccountSelection="false" @identicon-selected="_onAddressSelected">
+            </IdenticonSelector>
+            <Loader v-if="state === 'finished'" state="success" title="Address Added"></Loader>
+        </SmallPage>
+
+        <button class="global-close nq-button-s" @click="close">
+            <span class="nq-icon arrow-left"></span>
+            Back to {{request.appName}}
+        </button>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Vue } from 'vue-property-decorator';
+import { PageBody, SmallPage } from '@nimiq/vue-components';
+import LedgerUi from '../components/LedgerUi.vue';
+import Loader from '../components/Loader.vue';
+import IdenticonSelector from '../components/IdenticonSelector.vue';
+import { Static } from '../lib/StaticStore';
+import { ParsedSimpleRequest } from '../lib/RequestTypes';
+import { Address } from '../lib/PublicRequestTypes';
+import { WalletInfo } from '../lib/WalletInfo';
+import { AccountInfo } from '../lib/AccountInfo';
+import LedgerApi from '../lib/LedgerApi';
+import { WalletStore } from '../lib/WalletStore';
+import { ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, ERROR_CANCELED } from '../lib/Constants';
+import LabelingMachine from '../lib/LabelingMachine';
+
+@Component({components: {PageBody, SmallPage, LedgerUi, Loader, IdenticonSelector}})
+export default class AddAddressLedger extends Vue {
+    private static readonly State = {
+        LEDGER_INTERACTION: 'ledger-interaction', // can be instructions to connect or also display of an error
+        IDENTICON_SELECTION: 'identicon-selection',
+        FINISHED: 'finished',
+    };
+
+    @Static private request!: ParsedSimpleRequest;
+
+    private state: string = AddAddressLedger.State.LEDGER_INTERACTION;
+    private account!: WalletInfo;
+    private addressesToSelectFrom: AccountInfo[] = [];
+
+    private async created() {
+        // called every time the router shows this page
+        const account = await WalletStore.Instance.get(this.request.walletId);
+        if (!account) {
+            this.$rpc.reject(new Error('Account ID not found'));
+            return;
+        }
+        this.account = account;
+
+        let startIndex = 0;
+        const newestAddress = Array.from(account.accounts.values()).pop();
+        if (newestAddress) {
+            const newestIndex = LedgerApi.getKeyIdForBip32Path(newestAddress.path);
+            startIndex = (newestIndex !== null ? newestIndex : -1) + 1;
+        }
+
+        const pathsToDerive = [];
+        for (let keyId = startIndex; keyId < startIndex + ACCOUNT_MAX_ALLOWED_ADDRESS_GAP; ++keyId) {
+            pathsToDerive.push(LedgerApi.getBip32PathForKeyId(keyId));
+        }
+
+        const derivedAddressInfos = await LedgerApi.deriveAccounts(pathsToDerive, this.request.walletId);
```

Comment:
Oh, the new implementations that already use `account/address` instead of `wallet/account` will fuck us so bad when we want to bulk-refactor the code base... :(

But that is a problem in all recently added code...

Reply:
Should be ok, as these variables don't need to be renamed anymore